source/simulator/core.cc

```diff
@@ -1617,28 +1621,23 @@ namespace aspect
       constraints.distribute (old_distributed_system);
       old_solution = old_distributed_system;
 
+      // do the same as above also for the free surface solution
       if (parameters.free_surface_enabled)
         {
           constraints.distribute (distributed_mesh_velocity);
           free_surface->mesh_velocity = distributed_mesh_velocity;
-        }
 
-      // do the same as above also for the free surface solution
-      if (parameters.free_surface_enabled)
-        {
-          LinearAlgebra::Vector distributed_mesh_vertices;
-          distributed_mesh_vertices.reinit(free_surface->mesh_locally_owned,
-                                           mpi_communicator);
+          LinearAlgebra::Vector distributed_mesh_displacements;
+
+          distributed_mesh_displacements.reinit(free_surface->mesh_locally_owned,
+                                                mpi_communicator);
```

Comment:
same here -- could probably be handled via a signal.

But that's for a separate patch.